plugins/package-managers/cargo/src/main/kotlin/Cargo.kt

```diff
@@ -191,14 +192,30 @@ class Cargo(
         )
 
         val nonProjectPackages = packageById.values.mapNotNullTo(mutableSetOf()) { cargoPkg ->
-            cargoPkg.takeUnless { it.isProject() }?.toPackage(hashes)
+            cargoPkg.takeUnless { it.isProject(workingDir) }?.toPackage(hashes)
         }
 
         return listOf(ProjectAnalyzerResult(project, nonProjectPackages))
     }
 }
 
-private fun CargoMetadata.Package.isProject() = source == null
+/**
+ * Return the local path for this Cargo package if applicable, or null if the Cargo package is not local.
```

Comment:
> Is this always an absolute path

Yes, because it's derived from `definitionFile` which is guaranteed to be absolute.

> should it be mentioned in the docs or even checked with an assertion?

I'm simply calling `absoluteFile` now to make it work in any case.

Reply:
> Yes, because it's derived from `definitionFile` which is guaranteed to be absolute.

My question was about the `File` returned by `getLocalPath()` not about the parent of the `definitionFile`.

Reply:
Yes, that's also guaranteed to be absolute (as it's derived from a URI).

Reply:
Would a `require` check make sense / mention in the function docs of `getLocalPath()` ?